### PR TITLE
Use legacy User-Agent by default, add more settings

### PR DESCRIPTION
**Pull request by felipeerias**
Our User-Agent by default (when not in Desktop mode) becomes:

    Mozilla/5.0 (Android 10; Mobile VR; rv:105.0) Gecko/105.0 Firefox/105.0

This can be modified through the settings:

- use Wolvic's own UA: report ourselves as "Firefox" (the GeckoView default) or "Wolvic"
- UA mode: "Mobile" or "Mobile VR"

User-Agent settings have been moved to the Developer section.

Add a new setting to decide whether to use Wolvic's own User-Agent or GeckoView's default one.

Add "Mobile" and "Mobile VR" variants to Wolvic's User-Agent, to match what GeckoView does.

Update the User-Agent when a page is reloaded, so changes to the settings can be effective immediately.

English, Spanish, and Galician translations.

### CODE REVIEW

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java [hunk 1]
```diff
@@ -219,9 +244,33 @@ private void setLocalAddon(boolean value, boolean doApply) {
...
+        mBinding.uaRadio.setOnCheckedChangeListener(mUaModeListener);
+
+        SettingsStore.getInstance(getContext()).setUaMode((Integer) mBinding.uaRadio.getValueForId(checkId));
+    }
+
+    @Override
+    public Point getDimensions() {
+        return new Point(WidgetPlacement.dpDimension(getContext(), R.dimen.settings_dialog_width),
+                WidgetPlacement.dpDimension(getContext(), R.dimen.developer_options_height));
+    }
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java:269]: I don't see the relationship of this with anything else. 

Is it something that was missing? Does it fix any particular issue? If that's the case it should go in a different PR.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java [hunk 2]
```diff
@@ -219,9 +244,33 @@ private void setLocalAddon(boolean value, boolean doApply) {
...
+
     @Override
     protected SettingViewType getType() {
-        return SettingViewType.LANGUAGE_VOICE;
+        return SettingViewType.DEVELOPER;
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java:273]: Same comment as above.

**Review by svillar — CHANGES_REQUESTED** [review]
Looking good, there are a couple of things that I'd like to clarify first.

Also, we talked about showing a dialog warning the user everytime they select a UA which is not the default one (in this case Wolvic's). That information dialog should tell users that enabling Wolvic UI's might break some sites and encourage them to report those missbehaviours. We could talk offline about the exact text to be used.

**Comment by svillar** [comment]
We should also check which the exact UA we want to use by default, because DelightXR videos cannot be played unless you append a `OculusBrowser/X` declaration in my own experience

File: app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java [hunk 3]
```diff
@@ -1135,18 +1138,24 @@ WResult<WAllowOrDeny> onLoadRequest(@NonNull WSession aSession, @NonNull LoadReq
...
 
-            // Here we set a default user agent if no override was found, BUT
-            // only if UA Mode is not Desktop, because the UA override
-            // takes precedence over mode overrides so if we set it, it will
-            // invalidate desktop mode UA. For VR and Mobile modes we don't change
-            // the UA so they are not affected.
-            if (mState.mSettings.getUserAgentMode() != WSessionSettings.USER_AGENT_MODE_DESKTOP &&
-                    userAgentOverride == null) {
-                userAgentOverride = DEFAULT_USER_AGENT;
+            Log.e(LOGTAG, "checking override: " + uri);
```
[the remark this answers is not in the document]
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java]: Done.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java (hunk 1 above)
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java:269]: This makes the dialog a bit taller, which is needed after adding the two settings for the User-Agent.

The alternative was to keep the dialog with the same height, but that looks worse:

08b9cbb2b6a8529c27325e8a27f43551

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java (hunk 2 above)
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java:273]: This is a little issue that I saw when changing the dialog, it seems too small for its own PR but I can open one if needed.

File: app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java [hunk 4]
```diff
@@ -1135,18 +1138,24 @@ WResult<WAllowOrDeny> onLoadRequest(@NonNull WSession aSession, @NonNull LoadReq
...
+            // in Desktop mode (which sets its own user agent). These settings define:
+            //  - UA mode: "Mobile" or "Mobile VR"
+            //  - use Wolvic UA: report ourselves as "Firefox" (the GeckoView default) or "Wolvic"
+            int mode = mState.mSettings.getUserAgentMode();
+            if (mode != WSessionSettings.USER_AGENT_MODE_DESKTOP) {
+                int defaultMode = SettingsStore.getInstance(mContext).getUaMode();
+                if (mode != defaultMode) {
+                    setUaMode(defaultMode);
+                    mode = defaultMode;
+                }
```
[the remark this answers is not in the document]
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java]: @svillar Done. I had to add some extra code so we don't reload all the tabs when the _"User-Agent mode"_ setting changes (because all of them get notified of the setting change).

The previous behaviour before all my changes was that you needed to open a new tab (i.e. create a new `Session`) to get the new UA value.

File: app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java [hunk 5]
```diff
@@ -76,7 +76,7 @@ public class Session implements WContentBlocking.Delegate, WSession.NavigationDe
     private static UriOverride sDesktopModeOverrides;
     private static final long KEEP_ALIVE_DURATION_MS = 1000; // 1 second.
 
-    private static final String WOLVIC_USER_AGENT_MOBILE = "Mozilla/5.0 (Android 10; Mobile; rv:105.0) Gecko/105.0 Wolvic/" + BuildConfig.VERSION_NAME;
+    private static final String WOLVIC_USER_AGENT_MOBILE = "Mozilla/5.0 (Android 10; Mobile; rv:105.0) Gecko/105.0 Wolvic/" + BuildConfig.VERSION_NAME + " Firefox/105.0 OculusBrowser/25.3.0.4.30.438623098 VR Safari/537.36";
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java:79]: Sorry for the back and forth, but it looks like we don't need that.

So we'll stick to the plan of having Gecko/x.x Firefox/x.x by default and then letting users to opt-in to Gecko/x.x Wolvic/x.x in developer options.

**Review by svillar — CHANGES_REQUESTED** [review]

**Comment by svillar** [comment]
Also as I mentioned in other comment, we should add some dialog warning the user about enabling Wolvic UA because it might break some sites.

File: app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java (hunk 5 above)
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java]: Change reverted. This is how the User-Agent works now:

- Not Wolvic's own UA, _VR_ mode (default): 
  `Mozilla/5.0 (Android 10; Mobile VR; rv:105.0) Gecko/105.0 Firefox/105.0`
- Not Wolvic's own UA, _Mobile_ mode:
  `Mozilla/5.0 (Android 10; Mobile; rv:105.0) Gecko/105.0 Firefox/105.0`
- Wolvic's own UA, _VR_ mode:
  `Mozilla/5.0 (Android 10; Mobile VR; rv:105.0) Gecko/105.0 Wolvic/1.2`
- Wolvic's own UA, _Mobile_ mode:
  `Mozilla/5.0 (Android 10; Mobile; rv:105.0) Gecko/105.0 Wolvic/1.2`

**Comment by felipeerias** [comment]
For reference, this is a related issue in Firefox Reality: https://github.com/MozillaReality/FirefoxReality/issues/478

**Comment by felipeerias** [comment]
I have added an alert dialog when you use Wolvic's own User-Agent:

> Warning
>
> Changing the default "User-Agent" value is an experimental feature and might break some websites.

**Comment by svillar** [comment]
Superseeded by #453
